FactoryWeather/App/VIP Home/HomePresenter.swift

```diff
@@ -42,12 +46,19 @@ private extension HomePresenter {
                                   visibleConditions: settings.conditions)
     }
     
-    func createHomeContentViewModel(withWeather weather: Weather) -> HomeContentView.ViewModel {
-        let settings = DataManager.getSettings()
-        let headerViewModel = createHeaderViewModel(withWeather: weather, usingSettings: settings)
-        let bodyViewModel = createBodyViewModel(withWeather: weather, usingSettings: settings)
-        return HomeContentView.ViewModel(skyGradient: SkyWeatherCondition.forIcon(weather.icon).gradient,
-                                         headerViewModel: headerViewModel,
-                                         bodyViewModel: bodyViewModel)
+    func createHomeContentViewModel(withWeather weather: Weather) -> Promise<HomeContentView.ViewModel> {
+        return DataManager.getSettings()
```

Comment:
The `presenter` should definitely not perform any `business` logic, but should rather get all necessary domain data from the `interactor`. 

With this in mind, the `presenter` should get the `settings` passed in to any relevant methods - in this particular case, the `presentWeather` method.